src/games/strategy/engine/lobby/client/login/CreateUpdateAccountPanel.java

```diff
@@ -122,9 +123,8 @@ public void actionPerformed(final ActionEvent e) {
     });
   }
 
-  @SuppressWarnings("deprecation")
   private void okPressed() {
-    if (!m_password.getText().equals(m_password2.getText())) {
+    if (!Arrays.equals(m_password.getPassword(),m_password2.getPassword())) {
```

Comment:
Good change. Another step can be taken, from the javadocs:
"For stronger security, it is recommended that the returned character array be cleared after use by setting each character to zero."